### ISSUE TITLE
Stocktakes: The ability to hide the theoretical quantity column

### ISSUE THREAD

**Issue by mark-prins** [openmsupply/mobile#2509]
## Is your feature request related to a problem? Please describe.

Part of the Côte d'Ivoire phase two requirements.

## Describe the solution you'd like

Ability to hide the snapshot quantity when editing a stocktake.
Somehow, only for CI? Configurable by store?

**Comment by joshxg** [comment]
I'm thinking just a custom data field `"hideSnapshotQuantity: true"` .. but this brings up again the topic of how best to do column customisations from your PR 🤣

**Comment by wlthomson** [comment]
I was thinking more along the lines of a customisation which allows the user to show/hide the snapshot quantity with a toggle. Could be wrong but this is how it's currently done on desktop?

**Comment by joshxg** [comment]
Is that under a pref? Seems easier to just hide the column - we also need to reclaim some space because of their excessive column additions !

**Comment by wlthomson** [comment]
I believe it's not under a pref, just under the IC customisation. I think it would be easier to hide the column, but the fact it can be toggled on desktop makes me think whether perhaps IC want to retain the ability for the user to view it if they want?

**Comment by joshxg** [comment]
`OBJECT SET VISIBLE(*;"obj_cb_show_quantity";user_permission_check (12))  // Show quantity` - is hidden without this user pref - the reason for the requirement is so that users counting stock don't just copy over the snapshot quantity (yes its stupid)

**Comment by wlthomson** [comment]
Hm, yeah but that's wrapped in `If (<>t_customization_name="Ivory Coast")`. Same with all the on click handling code... not sure why:

```
		If (<>t_customization_name="Ivory Coast")
			OBJECT SET VISIBLE(*;"stock_take_line_enter_quantity";(Self->=1))
			OBJECT SET VISIBLE(*;"stock_take_line_blank_quantity";(Self->#1))
			OBJECT SET ENTERABLE(*;"stock_take_line_enter_quantity";False)
			OBJECT SET ENTERABLE(*;"stock_take_line_enter_quantity";True)
		End if 
```

So looks like you can technically enable it with a pref, but if you are on IC... which seems weird since IC were the only ones who requested it.

**Comment by mark-prins** [comment]
was about to reply and say that it **is** under a pref. and technically it is. The implementation is also buried under the customisation, and I think that it shouldn't be.

for mobile then..? can we use the pref and hide the column?

Yes, the toggle matches desktop. However, IC haven't asked for a toggle in desktop, they only asked to hide the column. ( we did arrive at the toggle during a UAT session, so it is possible that they'd like it on mobile. All the same, I think we should just hide the column. that is the use-case we've been given: staff at a certain level aren't trusted to have access to current quantities when stock taking )

**Comment by wlthomson** [comment]
@mark-prins That makes it a lot easier so I'm good with that. Do you know where the pref is set? Is it per store or per user?

**Comment by mark-prins** [comment]
per user - page three of permissions, under stocktakes:

<img width="586" alt="Screen Shot 2020-03-09 at 3 30 59 PM" src="https://user-images.githubusercontent.com/9192912/76178090-1b4ea780-621b-11ea-9da0-db812cf20120.png">

**Comment by joshxg** [comment]
- We don't 'use' any prefs synced to mobile. And as far as I know, they're all store prefs. As far as I am aware, the user prefs are an array which makes it harder to 'just implement sync' for it.
- We are looking to implement the use of store prefs - but not done yet.
- That's why my suggestion is that for such a simple use case, we just use the custom data field.. I don't like it either - maybe this can spark conversation on using user prefs in mobile, as well as store prefs
- (We technically use the view/edit preferences user pref - I added into the body in the auth `isAdmin: true` when a user logs in by checking the array, so not technically 'sync')

**Comment by wlthomson** [comment]
Confirmed that pref checkbox only appears on IC customisation, so I think we can go ahead with a `custom_data` mapped to that pref? 

Long-term I think there is going to be some consolidation of `custom_data` and `[pref]`, but for now I think the two main options are to:

- map `custom_data` to `[pref]` (would have to be done on mobile sync?)
- sync `[pref]` to mobile (a bit premature, I think)

Thoughts?

**Comment by wlthomson** [comment]
Update:

@mark-prins @joshxg As of discussion, going to go ahead with the `custom_data` solution. Not great, but will get the job done for the shorter term and can be relatively easily incorporated into later mobile pref sync work.

**Comment by wlthomson** [comment]
Fixed by #2701.
